crates/rpc/engine/payload.rs

```diff
@@ -110,12 +116,48 @@ impl RpcHandler for NewPayloadV3Request {
             Err(ChainError::StoreError(_)) => Err(RpcErr::Internal),
             Ok(()) => {
                 info!("Block with hash {block_hash} executed succesfully");
+
+                storage.set_canonical_block(block.header.number, block_hash)?;
```

Comment:
we should only set a canonical block on forkchoiceUpdated.

Reply:
The problem im having with that I ran into some  hive tests that run the following pipeline:
```
engine_forkChoiceUpdatedV3 (with a new payload attributes) -> Here we build a new payload (a block with number head + 1) and return its id
engine_getPayloadV3 -> Here we return the payload created in the previous call
engine_newPayloadV3 -> Here we receive the payload (a block), execute it and add it to the chain
engine_getBlockByNumber -> Here we receive the block number for the block we created in engine_newPayloadV3, but as it is not a canonical block storage.get_block_header returns null
```

Reply:
Ideally we should have a way to fetch this block without it needing to be canonical, I could add a TODO to it so that we know it is a temporary solution we must remove in the future

Reply:
Added TODO along with issue explanation

Reply:
This is weird, maybe `getPayloadV3` sets the block as canonical. Let me look into the specs

Reply:
I have not seen it. I will create a ticket.

Reply:
https://github.com/lambdaclass/ethereum_rust/issues/589